### PR TITLE
Reactions

### PR DESCRIPTION
**Pull request by bago2k4**
Card: https://trello.com/c/PdMEE1yx

To test:
- open a topic view
- check that the reactions are there (they should have 0 - 0 - 0 count if it's clean)
- click on a reaction
- click on another
- click again on a reactions you already reacted
- make sure all end up in the state they should
- refresh the page and make sure the state is still the same
- click multiple times on a reaction and make sure the last state is preserved after a refresh

### CODE REVIEW

**Comment by belucid** [comment]
Reviewing.

**Comment by belucid** [comment]
@bago2k4 some odd behavior when you click (or unclick) more than 1 at the same time in quick succession. The original click temporarily reverts its color when you click the 2nd.

File: src/oc/web/actions.cljs [hunk 1]
```diff
@@ -685,11 +685,17 @@
   ; Reset flag to reload su list when needed
   (dissoc db :updates-list-loaded))
 
+(defn sort-reactions [entry]
+  (let [reactions (:reactions entry)
+        sorted-reactions (vec (sort #(compare (:reaction %1) (:reaction %2)) reactions))]
```
Comment (belucid) [inline src/oc/web/actions.cljs]: I believe `(sort :reaction reactions)` is sufficient.

File: src/oc/web/actions.cljs [hunk 2]
```diff
@@ -685,11 +685,17 @@
...
+    (assoc entry :reactions sorted-reactions)))
+
 (defmethod dispatcher/action :entries-loaded
   [db [_ {:keys [topic entries]}]]
   (let [fixed-entries (map #(utils/fix-topic % topic) (:items (:collection entries)))
+        with-sorted-reactions (vec (map #(sort-reactions %) fixed-entries))
         sort-pred (fn [a b] (compare (:created-at b) (:created-at a)))
-        sorted-fixed-entries (vec (sort sort-pred fixed-entries))]
+        sorted-fixed-entries (vec (sort sort-pred with-sorted-reactions))]
```
Comment (belucid) [inline src/oc/web/actions.cljs]: doesn't `(vec (sort :created-at with sorted-reactions))` work here?

File: src/oc/web/api.cljs [hunk 3]
```diff
@@ -824,4 +826,15 @@
...
                                                       :error (when-not success body)
                                                       :body (if (not (empty? body)) (json->cljs body) nil)
-                                                      :entry-uuid entry-uuid}]))))))
\ No newline at end of file
+                                                      :entry-uuid entry-uuid}]))))))
+
+(defn toggle-reaction
+  [topic-slug entry-uuid reaction-data]
+  (when (and topic-slug entry-uuid)
+    (let [reaction-link (or (utils/link-for (:links reaction-data) "react" "PUT")
```
Comment (belucid) [inline src/oc/web/api.cljs]: Might be good to have a version of `link-for` that's either `(utils/link-for links "react")` or `(utils/link-for links ["PUT" "DETELE"])`

**Comment by belucid** [comment]
@bago2k4 can't say I love having the reaction icon and count with some transparency when it's got reactions. Seems fine when they are at 0 so as not to overpower the rest of the entry. And seems fine for the background and border to react to hover by removing the transparency so it's clear it's an interactive element. But it seems a reaction with > 0 ought to have the icon and count w/o any transparency.

**Comment by bago2k4** [comment]
@belucid all should be addressed! Ready for your 👀 again!

**Pull request merged**